portal/models/user.py

```diff
@@ -1838,6 +1838,48 @@ def get_user_or_abort(uid, allow_deleted=False):
     return user
 
 
+def active_patients(
+        include_test_role=False, include_deleted=False,
+        require_orgs=None, filter_by_ids=None):
+    """Build query for active patients, filtered as specified
+
+    Common query for active (not deleted) patients.
+
+    :param include_test_role: Set true to include users with ``test`` role
+    :param include_deleted: Set true to include deleted users
+    :param require_orgs: Provide list of organization IDs if patients must
+        also have the respective UserOrganization association (different from
+        consents!)  User required to have at least one, not all orgs in given
+        ``require_orgs`` list.
+    :param filter_by_ids: List of user_ids to include in query filter
+    :return: Live SQL Alchemy query, for further filter additions or execution
+
+    """
+    patients_query = User.query.join(
+        UserRoles).filter(User.id == UserRoles.user_id).join(
+        Role).filter(Role.name == ROLE.PATIENT.value)
+
+    if not include_deleted:
+        patients_query = patients_query.filter(User.deleted_id.is_(None))
+
+    if not include_test_role:
+        test_user_ids = UserRoles.query.join(Role).filter(
+            UserRoles.role_id == Role.id).filter(
+            Role.name == ROLE.TEST.value).with_entities(UserRoles.user_id)
+        patients_query = patients_query.filter(
```

Comment:
This would be a more idiomatic approach (but probably still uses a subquery)
```
patients_query = patients_query.filter(
    ~User.roles.any(Role.name == ROLE.TEST.value)
)
```